src/integration/test_suites/smoke_test_fc_ratchet_seed_volatile_lock/smoke_test_fc_ratchet_seed_volatile_lock.c

```diff
@@ -0,0 +1,187 @@
+//********************************************************************************
+// SPDX-License-Identifier: Apache-2.0
+//
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//********************************************************************************
+#include <string.h>
+#include <stdint.h>
+#include <time.h>
+#include <stdlib.h>
+
+#include "soc_address_map.h"
+#include "printf.h"
+#include "riscv_hw_if.h"
+#include "soc_ifc.h"
+#include "caliptra_ss_lc_ctrl_address_map.h"
+#include "caliptra_ss_lib.h"
+#include "fuse_ctrl.h"
+#include "lc_ctrl.h"
+#include "fuse_ctrl_mmap.h"
+
+volatile char* stdout = (char *)SOC_MCI_TOP_MCI_REG_DEBUG_OUT;
+#ifdef CPT_VERBOSITY
+    enum printf_verbosity verbosity_g = CPT_VERBOSITY;
+#else
+    enum printf_verbosity verbosity_g = LOW;
+#endif
+
+/**
+   This tests sets the ratchet_seed_volatile_lock increasingly from all its possible value and checks which partitions are writable 
+ */
+
+// Tests the volatile lock by setting it to every possible value and writing/reading to each ratchet seed register
+
+void ratchet_seed_volatile_lock(void) {
+    uint32_t data[2];
+
+    VPRINTF(LOW, "==================================\n testing all ratchet seed registers: \n")
+    VPRINTF(LOW, "testing all ratchet seed registers. ratcher_seed_volatile_lock = 0 \n==================================\n\n")
+    // LOCK set to 0x0 - all partitions are writable
+    dai_wr(partitions[CPTRA_SS_LOCK_HEK_PROD_0].address, 0xFF, 0xFF, 32, 0);
```

Comment:
We should do a `dai_rd` to check what was written. Please note that due to the 32-bit granularity of this partition, only the first word will be written.